testinfra/test_ami.py

```diff
@@ -0,0 +1,307 @@
+aws_config:
+    creds:
+        enabled: false
+        check_frequency: 1h
+        refresh_buffer_duration: 6h
+"""
+pgsodium_root_key_content = (
+    "0000000000000000000000000000000000000000000000000000000000000000"
+)
+postgrest_base_conf_content = """
+db-uri = "postgres://authenticator:postgres@localhost:5432/postgres?application_name=postgrest"
+db-schema = "public, storage, graphql_public"
+db-anon-role = "anon"
+jwt-secret = "my_jwt_secret_which_is_not_so_secret"
+role-claim-key = ".role"
+openapi-mode = "ignore-privileges"
+db-use-legacy-gucs = true
+admin-server-port = 3001
+server-host = "localhost"
+db-pool-acquisition-timeout = 10
+max-rows = 1000
+db-extra-search-path = "public, extensions"
+"""
+gotrue_env_content = """
+API_EXTERNAL_URL=http://localhost
+GOTRUE_API_HOST=0.0.0.0
+GOTRUE_SITE_URL=
+GOTRUE_DB_DRIVER=postgres
+GOTRUE_DB_DATABASE_URL=postgres://supabase_auth_admin@localhost/postgres?sslmode=disable
+GOTRUE_JWT_ADMIN_ROLES=supabase_admin,service_role
+GOTRUE_JWT_AUD=authenticated
+GOTRUE_JWT_SECRET=my_jwt_secret_which_is_not_so_secret
+"""
+walg_config_json_content = """
+{
+  "AWS_REGION": "ap-southeast-1",
+  "WALG_S3_PREFIX": "",
+  "PGDATABASE": "postgres",
+  "PGUSER": "supabase_admin",
+  "PGPORT": 5432,
+  "WALG_DELTA_MAX_STEPS": 6,
+  "WALG_COMPRESSION_METHOD": "lz4"
+}
+"""
+anon_key = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImFhYWFhYWFhYWFhYWFhYWFhYWFhIiwicm9sZSI6ImFub24iLCJpYXQiOjE2OTYyMjQ5NjYsImV4cCI6MjAxMTgwMDk2Nn0.QW95aRPA-4QuLzuvaIeeoFKlJP9J2hvAIpJ3WJ6G5zo"
+service_role_key = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImFhYWFhYWFhYWFhYWFhYWFhYWFhIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTY5NjIyNDk2NiwiZXhwIjoyMDExODAwOTY2fQ.Om7yqv15gC3mLGitBmvFRB3M4IsLsX9fXzTQnFM7lu0"
+supabase_admin_key = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImFhYWFhYWFhYWFhYWFhYWFhYWFhIiwicm9sZSI6InN1cGFiYXNlX2FkbWluIiwiaWF0IjoxNjk2MjI0OTY2LCJleHAiOjIwMTE4MDA5NjZ9.jrD3j2rBWiIx0vhVZzd1CXFv7qkAP392nBMadvXxk1c"
+init_json_content = f"""
+{{
+  "jwt_secret": "my_jwt_secret_which_is_not_so_secret",
+  "project_ref": "aaaaaaaaaaaaaaaaaaaa",
+  "logflare_api_key": "",
+  "logflare_pitr_errors_source": "",
+  "logflare_postgrest_source": "",
+  "logflare_pgbouncer_source": "",
+  "logflare_db_source": "",
+  "logflare_gotrue_source": "",
+  "anon_key": "{anon_key}",
+  "service_key": "{service_role_key}",
+  "supabase_admin_key": "{supabase_admin_key}",
+  "common_name": "db.aaaaaaaaaaaaaaaaaaaa.supabase.red",
+  "region": "ap-southeast-1",
+  "init_database_only": false
+}}
+"""
+
+
+# scope='session' uses the same container for all the tests;
+# scope='function' uses a new container per test function.
+@pytest.fixture(scope="session")
+def host():
+    ec2 = boto3.resource("ec2", region_name="ap-southeast-1")
+    images = list(
+        ec2.images.filter(
+            Filters=[{"Name": "name", "Values": ["supabase-postgres-ci-ami-test"]}]
```

Comment:
We should be using the version string from the common.vars file here, yeah? Not sure which AMI this is currently set up to use?

Reply:
The test run will always publish an AMI with AMI name `supabase-postgres-ci-ami-test`. Subsequent runs will deregister older AMIs through the `force_deregister` option.

One alternative I haven't explored is to move the test run into a Packer [provisioning step](http://celien.info/posts/ami_packer_serverspec.html), that way we might not even need to publish the AMI. Can be done as a follow up.